cmd/juju/cloud/add_test.go

```diff
@@ -499,7 +501,47 @@ func (*addSuite) TestInteractiveAddCloud_PromptForNameIsCorrect(c *gc.C) {
 	// Running the command will return an error because we only give
 	// enough input to get to the prompt we care about checking. This
 	// test ignores this error.
-	command.Run(ctx)
+	err := command.Run(ctx)
+	c.Assert(errors.Cause(err), gc.Equals, io.EOF)
 
 	c.Check(out.String(), gc.Matches, "(?s).+Enter a name for your manual cloud: .*")
 }
+
+func (*addSuite) TestSpecifyingCloudFileThroughFlag_CorrectlySetsMemberVar(c *gc.C) {
+	command := cloud.NewAddCloudCommand(nil)
+	runCmd := func() {
+		testing.RunCommand(c, command, "garage-maas", "-f", "fake.yaml")
+	}
+	c.Assert(runCmd, gc.PanicMatches, "runtime error: invalid memory address or nil pointer dereference")
```

Comment:
The panic comes from the attempt to dereference the nil pointer for the `cloudMetadataStore`. We could construct an instance and pass it in, but then we'd have to ensure all the endpoints are stubbed out. All of that seemed tangential to what we're trying to test, so this seemed like the best way.

Maybe this is an indication that I'm actually doing more a full-stack test than I want. Maybe I should modify this to test `Init` is performing correctly. I think I'll do that.